src/graphql/graphql-api.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * GraphQL API
+ *
+ * @package vip-block-data-api
+ */
+
+namespace WPCOMVIP\BlockDataApi;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * GraphQL API to offer an alternative to the REST API.
+ */
+class GraphQLApi {
+	/**
+	 * Initiatilize the graphQL API, if its allowed
+	 *
+	 * @access private
+	 */
+	public static function init() {
+		$is_graphql_to_be_enabled = apply_filters( 'vip_block_data_api__is_graphql_enabled', true );
```

Comment:
True by default, but can be set to false if need be.

Reply:
If it is on by default, and a site has WP GraphQL enabled what happens to the existing JSON endpoint? Does it still return JSON?

Reply:
Yip, the existing endpoint will work just as it previously did. While testing this, I used the regular REST endpoint in tandem as well.